pkgs/applications/misc/activitywatch/default.nix

```
@@ -0,0 +1,49 @@
+{ config, lib, stdenv, fetchzip, qt6, wrapGAppsNoGuiHook, glib-networking, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "ActivityWatch";
+  version = "0.12.1";
+  meta = with lib; {
+    description = "ActivityWatch is an app that automatically tracks how you spend time on your devices.";
+    longDescription = ''
+      ActivityWatch is an app that automatically tracks how you spend time on your devices.
+      It is open source, privacy-first, cross-platform, and a great alternative to services like RescueTime, ManicTime, and WakaTime.
+      It can be used to keep track of your productivity, time spent on different projects, bad screen habits, or just to understand how you spend your time.
+    '';
+    homepage = "https://github.com/ActivityWatch/activitywatch";
+    license = licenses.mpl20;
+    platforms = platforms.linux;
+  };
+
+  src = fetchzip {
+    url = "https://github.com/ActivityWatch/activitywatch/releases/download/v${version}/activitywatch-v${version}-linux-x86_64.zip";
+    hash = "sha256-VdnAKcLQBWBVZqRyD1xoYuAfvTeCtcFJp3/slkQ6ynk=";
+  };
+
+  buildInputs = [ qt6.qtbase glib-networking ];
+  nativeBuildInputs = [ autoPatchelfHook wrapGAppsNoGuiHook ];
+  dontWrapQtApps = true;
+
+  installPhase = ''
+    installed=$out/opt/activitywatch
+    mkdir -p $installed
+    mv * $installed
+    mkdir -p $out/bin
+
+    # Symlink executables to /bin
+    ln -s $installed/aw-qt $out/bin/aw-qt
+    for name in "aw-server" "aw-server-rust" "aw-watcher-afk" "aw-watcher-window"; do
+        ln -s $installed/$name/$name $out/bin/$name
```

```suggestion
      ln -s $installed/$name/$name $out/bin/$name
```
in nix code we use 2 spaces even for nested shellcode